app/apollo/resolvers/cluster.js

```diff
@@ -386,12 +427,38 @@ const clusterResolvers = {
       try {
         logger.debug({req_id, user, org_id}, `${queryName} enter`);
 
-        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context);
+        let allAllowed = false;
+        try {
+          logger.info({req_id, user, org_id}, `${queryName} validating`);
+          await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context);
+          allAllowed = true;
+        }
+        catch(e){ // If exception thrown, user does NOT have auth to all resources of this type, and code must later filter based on fine grained auth
+        }
+        logger.info({req_id, user, org_id, allAllowed}, `${queryName} validating - allAllowed: ${allAllowed}`);
+
         const conditions = await getGroupConditionsIncludingEmpty(me, org_id, ACTIONS.READ, 'uuid', queryName, context);
 
-        const results = await models.Cluster.aggregate([
+        const searchFilter = {
+          org_id,
+          updated: { $gte: new Moment().subtract(1, 'day').toDate() },
+          ...conditions
+        };
+
+        let clusters = await commonClusterSearch(models, searchFilter, { limit: 0 });
+
+        if (!allAllowed){
+          clusters = await filterResourcesToAllowed(me, org_id, ACTIONS.READ, TYPES.CLUSTER, clusters, context);
+          logger.info({req_id, user, org_id, clusters}, `${queryName} filtered resources to allowed`);
+        }
+
+        // Extract the IDs of filtered authorzied clusters
+        const clusterIds = clusters.map(cluster => cluster._id);
+
+        let results = await models.Cluster.aggregate([
```

Comment:
As it would be wrong/unexpected to redefine `results`:
```suggestion
        const results = await models.Cluster.aggregate([
```